sdk/rust/examples/frontend/src/main.rs

```diff
@@ -0,0 +1,42 @@
+use eyre::Result;
+use dagger_sdk::{Container, Directory, Query};
+
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let client = dagger_sdk::connect().await?;
+    let build_directory = build_frontend(&client).await;
+    let image = build_prod_image(&client, build_directory).await;
+    let image_reference = push_image(image).await?;
+    println!("Image published at: {}", image_reference);
+    Ok(())
+}
+
+async fn build_frontend(client: &Query) -> Directory {
+    let backend_directory = client.host().directory("leptos-frontend");
+    client
+        .container()
+        .from("rust:latest")
+        .with_directory("./frontend", backend_directory)
+        .with_workdir("/frontend")
+        .with_exec(vec!["rustup", "target", "add", "wasm32-unknown-unknown"])
```

Comment:
we should add dependencies and such before importing the dir. Otherwise we will cache bust everything on a file change